src/build.rs

```diff
@@ -387,24 +411,70 @@ fn create_disk_image(
     output.write_all(&bootloader_data).context("Could not write output bootimage file")?;
     output.write_all(&kernel_info_block).context("Could not write output bootimage file")?;
 
-    // write out kernel elf file
-    let kernel_size = kernel.metadata()?.len();
-    let mut buffer = [0u8; 1024];
-    loop {
-        let (n, interrupted) = match kernel.read(&mut buffer) {
-            Ok(0) => break,
-            Ok(n) => (n, false),
-            Err(ref e) if e.kind() == io::ErrorKind::Interrupted => (0, true),
-            Err(e) => Err(e)?,
-        };
-        if !interrupted {
-            output.write_all(&buffer[..n])?
+    fn write_file_to_file(output: &mut File, datafile: &mut File) -> Result<usize, Error> {
+        let data_size = datafile.metadata()?.len();
+        let mut buffer = [0u8; 1024];
+        let mut acc = 0;
+        loop {
+            let (n, interrupted) = match datafile.read(&mut buffer) {
+                Ok(0) => break,
+                Ok(n) => (n, false),
+                Err(ref e) if e.kind() == io::ErrorKind::Interrupted => (0, true),
+                Err(e) => Err(e)?,
+            };
+            if !interrupted {
+                acc += n;
+                output.write_all(&buffer[..n])?
+            }
         }
+
+        assert!(data_size == acc as u64);
+
+        Ok(acc)
+    }
+
+    // fn pad_file_512(output: &mut File, written_size: usize) -> Result<(), Error> {
```

Comment:
Is this commented out function still needed?

Reply:
Nope, both of the commented parts can be removed.

Reply:
Ok, could you remove them then?

Reply:
Yep, will do when I get back to a computer.